agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -0,0 +1,292 @@
+package ai.koog.agents.features.opentelemetry.feature
+
+import ai.koog.agents.core.agent.context.element.getAgentRunInfoElement
+import ai.koog.agents.core.agent.context.element.getNodeInfoElement
+import ai.koog.agents.core.agent.entity.AIAgentStorageKey
+import ai.koog.agents.core.feature.AIAgentFeature
+import ai.koog.agents.core.feature.AIAgentPipeline
+import ai.koog.agents.core.feature.InterceptContext
+import ai.koog.agents.features.opentelemetry.feature.span.*
+import io.opentelemetry.api.trace.StatusCode
+import kotlinx.coroutines.currentCoroutineContext
+
+/**
+ * Represents the OpenTelemetry integration feature for tracking and managing spans and contexts
+ * within the AI Agent framework. This class manages the lifecycle of spans for various operations,
+ * including agent executions, node processing, LLM calls, and tool calls.
+ */
+public class OpenTelemetry {
+
+    /**
+     * Companion object implementing the AIAgentFeature interface to provide OpenTelemetry
+     * specific functionality for agents. It manages spans and contexts to trace and monitor
+     * the lifecycle of agent executions, nodes, LLM calls, and tool invocations.
+     *
+     * This class handles:
+     * - Initialization and configuration of OpenTelemetry agents.
+     * - Interception and tracing of agent lifecycle events such as agent start, finish,
+     *   run errors, and various activities like node execution, LLM calls, and tool calls.
+     * - Management of spans and contexts for monitoring and lifecycle completion.
+     *
+     * The implementation includes private utility methods for ensuring spans are handled
+     * correctly and resources are properly released.
+     */
+    public companion object Feature : AIAgentFeature<OpenTelemetryConfig, OpenTelemetry> {
+
+        override val key: AIAgentStorageKey<OpenTelemetry> = AIAgentStorageKey("agents-features-opentelemetry")
+
+        override fun createInitialConfig(): OpenTelemetryConfig {
+            return OpenTelemetryConfig()
+        }
+
+        override fun install(
+            config: OpenTelemetryConfig,
+            pipeline: AIAgentPipeline
+        ) {
+            val interceptContext = InterceptContext(this, OpenTelemetry())
```

Comment:
So the actual feature implementation is just empty? Maybe adding actual methods on the `OpenTelemetry` class itself and then moving the logic from interceptors to these methods and making interceptors invoke these methods would be a bit cleaner.